Added functionality to obtain circuit IDs

suhasks123 opened the pull request:
### Design:

TOR specifications mention that using *random numbers* to achieve the functionality is the optimal method. This might be for security reasons. But for the scale of our implementation, generating circuit IDs by *incrementing a preset number* is a simpler and more effective way of avoiding circuit ID collisions.

suhasks123 committed:
Added functionality to obtain circuit IDs

* onion_proxy/circuit.py: get_rand_circ_id() returns a circuit id while
creating a circuit.
* onion_proxy/main.py: Global variable current_circ_id takes care of circId
collisions by having get_rand_circ_id() increment it to generate a new circId.
* onion_router/onion_router.py: get_rand_circ_id() achieves similar functionality
to its counterpart in onion_proxy/circuit.py

Finished issue #5

NJnisarg commented:
@suhasks123 The master has diverged since you PR. Can you align it before we merge it?

suhasks123 committed:
Added functionality to obtain circuit IDs

* onion_proxy/circuit.py: get_rand_circ_id() returns a circuit id while
creating a circuit.
* onion_proxy/main.py: Global variable current_circ_id takes care of circId
collisions by having get_rand_circ_id() increment it to generate a new circId.
* onion_router/onion_router.py: get_rand_circ_id() achieves similar functionality
to its counterpart in onion_proxy/circuit.py

Finished issue #5

suhasks123 committed:
Merge branch 'master' into getCircId

suhasks123 commented:
> @suhasks123 The master has diverged since you PR. Can you align it before we merge it?

I resolved the merge conflicts

arpi-r commented on onion_proxy/main.py:
```diff
@@ -11,14 +11,17 @@
-	circ_id = Circuit.get_rand_circ_id()
+	global current_circ_id
+	circ_id = Circuit.get_rand_circ_id(current_circ_id)
+	current_circ_id += 1
```
Instead of incrementing current_circ_id again, set it to circ_id

NJnisarg replied:
@suhasks123 Please use something like a class variable and only one variable. It should not be this hard.

suhasks123 replied:
@NJnisarg So basically remove the get_rand_circ_id() altogether from circuit.py?

suhasks123 replied:
@arpi-r I made the changes

suhasks123 committed:
Merge branch 'getCircId' of https://github.com/suhasks123/tor_project into getCircId
